Source/BSGConfigurationBuilder.h

```diff
@@ -0,0 +1,16 @@
+#import <Foundation/Foundation.h>
+
+@class BugsnagConfiguration;
+
+@interface BSGConfigurationBuilder : NSObject
+
+/**
+ * Creates a configuration, applying supported options before returning the new
+ * config object. Unsupported options warn.
+ *
+ * @return a new BugsnagConfiguration object or nil if the a valid object could
```

Comment:
nit: I don't see nil being returned or warnings emitted.

Reply:
I've removed this part of the doc as that behaviour would be inconsistent with Android/the spec, and doesn't reflect reality.